scripts/verify.js

```diff
@@ -42,23 +56,110 @@ function isBinance() {
   return /bsc/.test(network);
 }
 
+function isPolygon() {
+  const network = hardhat.network.name
+  return /polygon/.test(network) || /matic/.test(network)
+}
+
 function etherscanApiKey() {
   if (isBinance()) {
+    info(`Using bsc scan api key`)
     return process.env.BSCSCAN_API_KEY
-  } else {
+  } 
+  else if(isPolygon()){
+    info(`Using polygonscan api key`)
+    return process.env.POLYGONSCAN_API_KEY
+  }
+  else {
     return process.env.ETHERSCAN_API_KEY
   }
 }
 
+
+/*
+needs to be in form: ardhat verify --config hardhat.config.polygon.js --network matic 0x5effa0823e486A5ED1D49d88A1374Fc337e1f9F4 
+  --contract contracts/builders/PoolWithMultipleWinnersBuilder.sol:PoolWithMultipleWinnersBuilder 
+"0x20F29CCaE4c9886964033042c6b79c2C4C816308" "0x41122Ca50202d13c809dfE88F60Da212A1525Ed7" "0x4d1639e4b237BCab6F908A1CEb0995716D5ebE36" 
+"0xaFcEa072BcBad91029A2bA0b37bAC8269dd4f5E6" "0x72Edd573E230C7d68274Bf718A4C6aD82b5d5f90"
+*/
+async function verifyPolygonScan(){
```

Comment:
What is all this?  Could we have more re-use code here?  I would love to see it cleaned up as much as humanly possible